ios/TelegramShare.m

```diff
@@ -27,7 +27,7 @@ - (void)shareSingle:(NSDictionary *)options
     NSURL * shareURL = [NSURL URLWithString:urlTelegram];
     
     if ([[UIApplication sharedApplication] canOpenURL: shareURL]) {
-        [[UIApplication sharedApplication] openURL: shareURL];
+       [[UIApplication sharedApplication] openURL:shareURL options:@{} completionHandler:^(BOOL success) {}];
```

Comment:
shouldn't the `resolve(@[@true, @""]);` be in the completion handler ? so that the promise resolves when the share is complete

Reply:
Thanks for the feedback, you're right. First i did it like that and then i checked how its handled for other share options such as Instagram in **InstagramShare.m** : 
`if ([[UIApplication sharedApplication] canOpenURL: shareURL]) {
        [[UIApplication sharedApplication] openURL:shareURL options:@{} completionHandler:nil];
        resolve(@[@true, @""]);
    }` 

I didn't wanted to implement different logic from the other sharing options, but i can change it like this if its okay:
```
if ([[UIApplication sharedApplication] canOpenURL:shareURL]) {
        [[UIApplication sharedApplication] openURL:shareURL options:@{} completionHandler:^(BOOL success) {
            if (success) {
                resolve(@[@true, @"Successfully opened Telegram"]);
            } else {
                NSString *errorMessage = @"Failed to open Telegram for an unknown reason";
                NSError *error = [NSError errorWithDomain:@"com.rnshare" code:2 
                                                userInfo:@{NSLocalizedDescriptionKey: errorMessage}];
                reject(@"Open Error", errorMessage, error);
            }
        }];
    } 
```

But i do not know in which cases completionHandler returns false with the successfull canOpenURL condition. (Maybe this case happens if the Telegram crashes because of some reason?)

Reply:
your proposed new code *seems* more correct. I'm not sure why the other ones aren't doing that but if they are not, *and* your code works well for you (testing expected success and failure modes) *then* it seems like your code should be the style we use here and likely everywhere...